### PR TITLE
Added cli and yml parser

### PR DESCRIPTION
**Pull request by ricomnl**
#### What does this implement/fix? Explain your changes
- Adds a simple starter cli to read a pdb, convert it into a graph, and save it as a pickle. New features can be added in the future.
- Implements a yaml parser so that config objects can be written as yaml and will be parsed and read as Pydantic Objects. The yaml parser allows the specification of classes, functions and partial functions
- Because Python doesn't have a proper __eq__ operator to compare that two partial functions are the same, I added an __eq__ overwrite operator for `ProteinGraphConfig` which uses `deepdiff` to check whether two Pydantic objects are equal

#### What testing did you do to verify the changes in this PR?
- A test that confirms that the cli produces valid outputs.
- A test that confirms that a Pydantic object parsed from a yaml file is the same as a Pydantic object that was initiated within python

### CODE REVIEW

File: environment-dev.yml [hunk 1]
```diff
@@ -70,6 +70,8 @@ dependencies:
 - versioneer
 - python-wget
 - xarray
+- deepdiff
```
Comment (a-r-j) [inline environment-dev.yml:73]: We do dependency management in [these files](https://github.com/a-r-j/graphein/tree/master/.requirements)

File: graphein/protein/config.py [hunk 2]
```diff
@@ -6,16 +6,41 @@
...
+from deepdiff import DeepDiff
+from deepdiff.operator import BaseOperator
 from pydantic import BaseModel
 from typing_extensions import Literal
 
 from graphein.protein.edges.distance import add_peptide_bonds
 from graphein.protein.features.nodes.amino_acid import meiler_embedding
 
 
+def partial_functions_equal(func1, func2):
```
Comment (a-r-j) [inline graphein/protein/config.py:22]: Some docstrings would be super helpful here

File: graphein/cli.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        pdb_paths = [pdb_path]
+    elif pdb_path.is_dir():
+        pdb_paths = [pdb for pdb in pdb_path.glob("*.pdb")]
+    else:
+        raise NotImplementedError(
+            "Given PDB path needs to point to either a pdb file or a directory with pdb files."
+        )
+
+    for path in pdb_paths:
+        g = construct_graph(config=config, pdb_path=str(path))
```
Comment (a-r-j) [inline graphein/cli.py:54]: Note to self: replace with multiprocessing graph constructor when `dataloaders` PR is finished

**Review by a-r-j — COMMENTED** [review]
LGTM!

File: graphein/protein/config.py (hunk 2 above)
Comment (ricomnl) [reply graphein/protein/config.py:22]: Done

**Pull request merged**